src/login/logind-session.c

```diff
@@ -1232,6 +1235,10 @@ bool session_may_gc(Session *s, bool drop_not_started) {
                         return false;
         }
 
+        if (s->leader_pidfd > 0 && s->leader > 0)
+                if (pid_is_alive(s->leader))
```

Comment:
please add pidfd_is_alive() that first translates pidfd to pid internally, and then does the pid_is_alive() check via the pid, and then checks if the pid still matches the pidfd afterwards, before returning the result. So that the pid cannot be reused while we operate on it